File: Missing Numbers.py

def diff(A, B):
    i = 0
    L = {}

    for i in range(len(A)):
        if B[i] not in L:
            L[B[i]] = 1
        else:
            L[B[i]] += 1
            
        if A[i] not in L:
            L[A[i]] = -1
        else:
            L[A[i]] -= 1
    
    for i in range(len(A), len(B)):
        if B[i] not in L:
            L[B[i]] = 1
        else:
            L[B[i]] += 1

    #print(L)
    for x in L.keys():
        if L[x]:
            print(x, end = ' ')
        
if __name__ == '__main__':
    str = input()
    A = [int(x) for x in input().split()]
    str = input()
    B = [int(x) for x in input().split()]
    diff(A, B)

File: Missing Numbers.java
import java.io.*;
import java.util.*;
import java.text.*;
import java.math.*;
import java.util.regex.*;

public class Solution {

    public static void main(String[] args) {
        /* Enter your code here. Read input from STDIN. Print output to STDOUT. Your class should be named Solution. */
        try{
		BufferedReader br = 
                      new BufferedReader(new InputStreamReader(System.in));
            int n = Integer.parseInt(br.readLine());
            String[] arrA = br.readLine().split(" ");
            int m = Integer.parseInt(br.readLine());
            String[] arrB = br.readLine().split(" ");
            
            int[] A = new int[n];
            int[] B = new int[m];
            HashMap<Integer,Integer> hA = new HashMap<Integer,Integer>();
            HashMap<Integer,Integer> hB = new HashMap<Integer,Integer>();
            ArrayList<Integer> missing = new ArrayList<Integer>(); 
            for(int i=0; i<n; i++)
            {
                A[i] = Integer.parseInt(arrA[i]);
                if(!hA.containsKey(A[i]))
                {
                    hA.put(A[i],1);
                }
                else
                {
                    hA.put(A[i], hA.get(A[i])+1);
                }
            }
            
            for(int i=0; i<m; i++)
            {
                B[i] = Integer.parseInt(arrB[i]);
                if(!hB.containsKey(B[i]))
                {
                    hB.put(B[i],1);
                }
                else
                {
                    hB.put(B[i], hB.get(B[i])+1);
                }
            }
            
            Iterator iterB = hB.entrySet().iterator();
            while(iterB.hasNext())
            {
                Map.Entry mapEntry = (Map.Entry) iterB.next();
                if(hA.containsKey(mapEntry.getKey()) && (Integer.parseInt(mapEntry.getValue().toString())-hA.get(mapEntry.getKey())==0))
                   {
                       
                   }
                else
                {
                    missing.add(Integer.parseInt(mapEntry.getKey().toString()));
                }
            }
            Collections.sort(missing);
            for(int i=0; i<missing.size(); i++)
            {
                System.out.print(missing.get(i)+" ");
            }
            
        }catch(IOException io)
        {
            io.printStackTrace();
        }
    }
}